### ISSUE TITLE
[Bug]: Callout is not closed when focus moves out from it

### ISSUE THREAD

**Issue by RanaKh1** [microsoft/fluentui#23730]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
Window 11, Browser- Chrome
```

### Are you reporting Accessibility issue?

yes

### Reproduction

https://codepen.io/rana-khell/pen/XWErmyL

## Actual Behavior
when focus moves out of the callout, the callout isn't closed.
![image](https://user-images.githubusercontent.com/86226959/175802564-1099d765-e202-407d-b26a-c9dd59084255.png)

Clicking tab will let the focus land outside of the callout (on Console button in the example above), however this does not close the callout.

## Expected Behavior
Callout should be closed (dismissed). As requested by our accessibility team, focus should remain inside the callout, until it gets closed.

### Requested priority

Normal

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by gouttierre** [comment]
@RanaKh1 - Thanks for filing this issue with us. To set expectations the developers will review this issue once capacity allows.

@TristanWatanabe  - Would you be able to confirm if this is a regression, or if this behavior is an issue? Thank you

**Comment by TristanWatanabe** [comment]
@RanaKh1 This behavior is by design as the `Callout` is only meant to be dismissed via clicking/pressing the trigger when the callout is open, via pressing the escape key when the callout has focus or via mouseclick outside of the callout. If you want focus to remain inside until it's closed, I suggest using the `FocusTrapCallout` variant instead.
